images/prow-tests/Dockerfile

```diff
@@ -100,57 +140,58 @@ ARG KUBETEST2_VERSION=ea3c260668075a29dd2b7f3ba4ddf1fe78bdf898
+ARG PROTOC_GEN_GOGOFASTER_VERSION=v1.3.1
+ARG GOTESTSUM_VERSION=v1.7.0
+ARG KAIL_VERSION=v0.15.0
+ARG LICHE_VERSION=v0.0.0-20200229003944-f57a5d1c5be4
+ARG GO_LICENSES_VERSION=v1.0.0
+ARG DOCKER_CREDENTIAL_GCR_VERSION=v2.0.5
+
+# Extra tools through go install
 # These run using the golang image version of Go, not any defined by `gvm`
-RUN go get github.com/google/ko@${KO_VERSION}
-RUN go get github.com/boz/kail/cmd/kail
-RUN go get gotest.tools/gotestsum
-RUN go get github.com/raviqqe/liche@v0.0.0-20200229003944-f57a5d1c5be4  # stable liche version for checking md links
-RUN go get sigs.k8s.io/kind@${KIND_VERSION}
-RUN go get sigs.k8s.io/kubetest2@${KUBETEST2_VERSION}
-RUN go get sigs.k8s.io/kubetest2/kubetest2-gke@${KUBETEST2_VERSION}
-RUN go get sigs.k8s.io/kubetest2/kubetest2-kind@${KUBETEST2_VERSION}
-RUN go get sigs.k8s.io/kubetest2/kubetest2-tester-exec@${KUBETEST2_VERSION}
-RUN go get google.golang.org/protobuf/cmd/protoc-gen-go@${PROTOC_GEN_GO_VERSION}
-RUN go get github.com/google/go-licenses
+RUN go install github.com/google/ko@${KO_VERSION}
+RUN go install github.com/boz/kail/cmd/kail@${KAIL_VERSION}
+RUN go install gotest.tools/gotestsum@${GOTESTSUM_VERSION}
+RUN go install github.com/raviqqe/liche@${LICHE_VERSION}  # stable liche version for checking md links
+RUN go install sigs.k8s.io/kind@${KIND_VERSION}
+RUN go install sigs.k8s.io/kubetest2@${KUBETEST2_VERSION}
+RUN go install sigs.k8s.io/kubetest2/kubetest2-gke@${KUBETEST2_VERSION}
+RUN go install sigs.k8s.io/kubetest2/kubetest2-kind@${KUBETEST2_VERSION}
+RUN go install sigs.k8s.io/kubetest2/kubetest2-tester-exec@${KUBETEST2_VERSION}
+RUN go install google.golang.org/protobuf/cmd/protoc-gen-go@${PROTOC_GEN_GO_VERSION}
+RUN go install github.com/google/go-licenses@${GO_LICENSES_VERSION}
+RUN go install sigs.k8s.io/kind@${KIND_VERSION}
+
 
 # According to https://github.com/knative/test-infra/pull/2762 protoc-gen-gogofaster is unsupported (and shouldn't be used?)
 # Not sure when it can be removed, maybe knative-0.26?
-RUN go get github.com/gogo/protobuf/protoc-gen-gogofaster@v1.3.1
-
-# TODO(chizhg): remove it once we replace with kntest in the places where this tool is used
-# Install our own tools
-RUN go get knative.dev/pkg/testutils/junithelper@release-0.17
+RUN go install github.com/gogo/protobuf/protoc-gen-gogofaster@${PROTOC_GEN_GOGOFASTER_VERSION}
 
 # Docker
-RUN go get -u github.com/GoogleCloudPlatform/docker-credential-gcr@v2.0.5
-
-############################################################
-FROM stable-stuff AS test-infra-builds
+RUN go install github.com/GoogleCloudPlatform/docker-credential-gcr@${DOCKER_CREDENTIAL_GCR_VERSION}
 
 # We do this instead of `go get` so the tools are locked to the Dockerfile's commit
 COPY . /go/src/knative.dev/test-infra
 
 # Build custom tools in the container
 
-RUN go install knative.dev/test-infra/tools/coverage
-RUN go install knative.dev/test-infra/kntest/cmd/kntest
+RUN go install knative.dev/test-infra/tools/coverage@latest
```

Comment:
The reason we didn't add the version is explained at L173 - `We do this instead of "go get" so the tools are locked to the Dockerfile's commit`. We also run the image build in presubmit so that makes sure the tool version we install is always consistent. Can we revert this change?

Reply:
I know but it fails when I don't add latest.

```
Removing intermediate container 4b7cf464eacf
 ---> 86c3a9d15be6
Step 60/69 : COPY . /go/src/knative.dev/test-infra
 ---> a088264a3284
Step 61/69 : RUN go install knative.dev/test-infra/tools/coverage
 ---> Running in e9acd904c7e9
go install: version is required when current directory is not in a module
        Try 'go install knative.dev/test-infra/tools/coverage@latest' to install the latest version
The command '/bin/sh -c go install knative.dev/test-infra/tools/coverage' returned a non-zero code: 1
ERROR
ERROR: build step 0 "gcr.io/cloud-builders/docker" failed: step exited with non-zero status: 1

```

Reply:
Changing it to `RUN cd /go/src/knative.dev/test-infra && go install ./tools/coverage` will work. Ditto for other commands.